localstack/services/opensearch/cluster.py

```diff
@@ -225,27 +326,60 @@ def _log_listener(self, line, **_kwargs):
         LOG.info("[%s] %s", self.port, line.rstrip())
 
 
-class CustomEndpoint:
+class EndpointProxy:
+    def __init__(self, base_url: str, forward_url: str, custom_endpoint: CustomEndpoint) -> None:
+        super().__init__()
+        self.base_url = base_url
+        self.forward_url = forward_url
+        self.custom_endpoint = custom_endpoint
+        self.routing_rules = None
+
+    def register(self):
+        _url = urlparse(self.base_url)
+        self.routing_rules = register_cluster(
+            host=_url.hostname,
+            path=_url.path,
+            forward_url=self.forward_url,
+            custom_endpoint=self.custom_endpoint,
+        )
+
+    def unregister(self):
+        try:
+            for rule in self.routing_rules:
+                ROUTER.remove_rule(rule)
+        except Exception:
+            LOG.debug("Router rule removing failed", exc_info=True)
```

Comment:
This seems a bit dangerous. Why is it necessary to catch all exceptions here?
From my perspective there are two possibilities:
- There is an issue with the control flow such that routes are somehow removed that have not been added or have been removed before.
  - In this case, we need to investigate and fix this (or describe why this is necessary and _only_ catch the errors which can be thrown).
- This is just a remnant from the refactoring and actually not necessary anymore.
  - In this case, the try except should be removed.

Reply:
Fixed :+1: there was a double unregister that caused trouble here